p8e-api-webservice/src/main/kotlin/io/provenance/p8e/webservice/config/AppConfig.kt

```diff
@@ -113,14 +113,16 @@ class AppConfig : WebMvcConfigurer {
     @Bean
     fun signAndVerifyApi(smartKeyApiClient: ApiClient): SignAndVerifyApi = SignAndVerifyApi(smartKeyApiClient)
 
+    @Bean
+    fun authenticationApi(smartKeyApiClient: ApiClient): AuthenticationApi = AuthenticationApi(smartKeyApiClient)
+
     @Bean
     fun securityObjectsApi(smartKeyApiClient: ApiClient): SecurityObjectsApi = SecurityObjectsApi(smartKeyApiClient)
 
     @Bean
-    fun smartKeySigner(signAndVerifyApi: SignAndVerifyApi, securityObjectsApi: SecurityObjectsApi): SmartKeySigner = SmartKeySigner(signAndVerifyApi, securityObjectsApi)
+    fun smartKeySigner(signAndVerifyApi: SignAndVerifyApi, securityObjectsApi: SecurityObjectsApi, authenticationApi: AuthenticationApi): SmartKeySigner
```

Comment:
@piercetrey-figure  Is a different SmartKeySigner instance created at WebService time? If so I'd imagine it's own instance would be able to refresh it's own token.

Reply:
the WebService isn't using the signer directly, just the securityObjectsApi (this bean is to satisfy the SignerFactory being injected into the AffiliateService I think), but it should be able to do its own token refreshing, so either it can call the SmartKeyAuthRefresh thing in the KeyManagementService, or maybe the SmartKeySigner is getting intialized anyways... though probably not as the factory would have to be called, right?

Reply:
cool, so the AuthRefresh is being setup during SmartKeySigner's init, though im guessing p8e-api doesn't have visibility to the KeyManagementService? Id rather have the AuthRefresh happen at one place.

Reply:
Yeah, I don't think p8e-api is using the KeyManagementService, though it could if needed. I think it is probably fine where it is, all I was saying is it could maybe be an entirely separate concern (separate from key management, or signing, just something that keeps the token alive for whatever might need it)